Integrations/src/main/java/io/deephaven/integrations/learn/Gatherer.java

```diff
@@ -0,0 +1,37 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.db.v2.sources.ColumnSource;
+
+public class Gatherer {
+
+    /**
+     * Copy data from a table into a 2d tensor.
+     *
+     * @param indexSet indices of the rows of the table to put into the tensor
+     * @param columnSources columns of data to put into the tensor
+     * @return contiguous RAM allocated for the tensor. When a numpy tensor is passed in for this argument, jpy will
+     *         handle passing the memory reference as a 1d java array here.
+     */
+    public static double[] create_2d_tensor(final IndexSet indexSet, final ColumnSource<?>[] columnSources) {
```

Comment:
1) since this is Java, should it be camel case?
2) What other arguments should be provided here?  e.g. transpose
3) Better name?  `tensor_buffer_2d`?
4) Support other output types